src/envoy/admin/manager/site_reading.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+from datetime import datetime
+
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from envoy.admin.crud.site_reading import (
+    count_site_readings_for_site_and_time,
+    select_agg_id_for_site,
+    select_csip_aus_site_type_ids,
+    select_site_readings_for_site_and_time,
+)
+from envoy.admin.mapper.site_reading import AdminSiteReadingMapper
+from envoy_schema.admin.schema.site_reading import CSIPAusSiteReadingUnit, CSIPAusSiteReadingPageResponse
+
+
+class AdminSiteReadingManager:
+    """Logic layer for admin site reading operations."""
+
+    @staticmethod
+    async def get_site_readings_for_site_and_time(
+        session: AsyncSession,
+        site_id: int,
+        csip_unit: CSIPAusSiteReadingUnit,
+        start_time: datetime,
+        end_time: datetime,
+        start: int = 0,
+        limit: int = 500,
+    ) -> CSIPAusSiteReadingPageResponse:
+        """Get site readings for specified site within a time range."""
+
+        # Convert CSIP unit to UOM for database query
+        uom = AdminSiteReadingMapper.csip_unit_to_uom(csip_unit)
+
+        # Query 1: get the aggregator id from site to help hit the index of SiteReadingType
+        aggregator_ids = await select_agg_id_for_site(session=session, site_id=site_id)
```

Comment:
This will raise an exception if the site_id DNE - this will manifest into a HTTP 500 from the client's perspective.

I would recommend using the admin/crud - select_single_site_no_scoping function and just handling the case when you get a None returned (either raise a LoggedHTTPException or return None and have the controller handle it - I have a slight preference towards the latter so that HTTP considerations aren't polluting the manager layer.

It would also make select_agg_id_for_site superfluous - which could then be deleted